### PR TITLE
Validating variable names

### PR DESCRIPTION
**Pull request by SethPoulsen**
Addresses issue #14

### CODE REVIEW

**Comment by jasonxia17** [comment]
I think we should also exclude certain C++ keywords. In particular, nullptr should be an invalid variable name because we'll probably want to be able to reassign a pointer to point to nullptr, which won't work if there's a variable named nullptr.

**Review by jasonxia17 — CHANGES_REQUESTED** [review]
I think we should also exclude certain C++ keywords. In particular, nullptr should be an invalid variable name because we'll probably want to be able to reassign a pointer to point to nullptr, which won't work if there's a variable named nullptr.

**Comment by SethPoulsen** [comment]
Great point! Actually, we should probably just have an exclusion list that has all keywords from common languages.

File: tsconfig.json [hunk 1]
```diff
@@ -5,7 +5,8 @@
         "removeComments": false,
         "preserveConstEnums": true,
         "outDir": ".build",
-        "allowJs": true
+        "allowJs": true,
+        "target": "es2015"
```
Comment (SethPoulsen) [inline tsconfig.json:9]: I did this so I could use `Set()`. I don't think any University students are going to be using browsers more than 4 years old, are they (all browsers supported the full es2015 spec by 2016).

File: src/VariableValidation.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    "nonlocal", "while", "assert", "del", "global", "not", "with", "async",
+    "elif", "if", "or", "yield",
+    // Other
+    "NULL", "null",
+]);
+
+export function validateVariableName(name: string, alreadyUsed: string[]): boolean {
+
+    if (!varNameRe.test(name)) {
+        alert("Variable names must start with a letter and contain only letters and numbers.");
```
Comment (jasonxia17) [inline src/VariableValidation.ts]: Should we mention underscores at all? e.g. "Variable names must start with a letter or underscore, and must contain only letters, numbers, and underscores."

Comment (SethPoulsen) [reply src/VariableValidation.ts]: I was going to leave it out to make the message more simple, but actually I think its fine to mention it. We can readjust later to user feedback when we start beta testing.

**Pull request merged**